libobs/obs-source.c

```diff
@@ -5434,7 +5491,11 @@ static float get_source_volume(obs_source_t *source, uint64_t os_time)
 
 	bool muted = !source->enabled || source->muted ||
 		     (source->push_to_mute_enabled && push_to_mute_active) ||
-		     (source->push_to_talk_enabled && !push_to_talk_active);
+		     (source->push_to_talk_enabled && !push_to_talk_active) ||
+		     (obs->audio.ignore_duplicate_audio &&
```

Comment:
I am confused by this check - `ignore_duplicate_audio` should never be set for a device that does not have `OBS_SOURCE_DO_NOT_SELF_MONITOR` set, as `update_audio_output_dup_refs` would return early without doing anything.

Also what is the difference between `user_ignore_duplicate_audio` and `ignore_duplicate_audio`? Is this essentially faking a user action for `apply_audio_action`?

Reply:
> Also what is the difference between user_ignore_duplicate_audio and ignore_duplicate_audio? Is this essentially faking a user action for apply_audio_action?

The ignore_duplicate_audio variable is used as async in the apply_audio_actions function. And user_ignore_duplicate_audio is used to check to if the audio is already ignored, so the update_audio_output_dup_refs function isn't ran if this is the case. This is also the case for muted and user_muted variables, for example.

Reply:
> I am confused by this check - ignore_duplicate_audio should never be set for a device that does not have OBS_SOURCE_DO_NOT_SELF_MONITOR set, as update_audio_output_dup_refs would return early without doing anything.

The ignore_duplicate_audio check is used for any audio source that is not OBS_SOURCE_DO_NOT_SELF_MONITOR. If duplicate audio is detected, for example in the media source,  the audio will not sent to the output and only be sent to the audio monitor.